### PR TITLE
Fixing Single product page

### PR DESCRIPTION
**Pull request by hassanidris**

### CODE REVIEW

File: client/src/components/ProductsList.jsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import useFetch from "../../hooks/useFetch";
 
 const ProductsList = () => {
   const { data, loading, error } = useFetch("/doors?populate=*");
+  console.log(data);
```
Comment (Quantinx) [inline client/src/components/ProductsList.jsx:7]: Missed console log

File: client/src/pages/Product/Product.jsx [hunk 2]
```diff
@@ -1,20 +1,45 @@
...
 
 const Product = () => {
-  const [selectedImg, setSelectedImg] = useState(0);
-  const data = ["../images/sng-pro1.jpg", "../images/sng-pro2.jpg"];
+  const id = useParams().id;
+  console.log(id);
+  const [selectedImg, setSelectedImg] = useState("");
+
+  const { data, loading, error } = useFetch(`/doors/${id}?populate=*`);
+  console.log(data);
```
Comment (Quantinx) [inline client/src/pages/Product/Product.jsx:13]: spooky log

File: client/src/pages/Product/Product.jsx [hunk 3]
```diff
@@ -1,20 +1,45 @@
...
+
+  useEffect(() => {
```
Comment (Quantinx) [inline client/src/pages/Product/Product.jsx:15]: Approved - potential for refactoring  - move this out of a useeffect

**Review by Quantinx — APPROVED** [review]
Couple of missed console logs but nothing actual problematic

File: client/src/components/ProductsList.jsx (hunk 1 above)
Comment (hassanidris) [reply client/src/components/ProductsList.jsx:7]: I will remove it in the next pull

**Pull request merged**